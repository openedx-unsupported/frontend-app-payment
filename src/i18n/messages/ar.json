{
  "payment.error.fetch.basket": "حدثت مشكلة غير متوقعة، إن استمرت المشكلة فيرجى {supportLink}.",
  "payment.error.fetch.basket.support.fragment": "الاتصال بالدعم",
  "payment.messages.enrollment-code-product-info.header": "أتشتري لنفسك فقط؟",
  "payment.messages.enrollment-code-product-info.body": "إن كنت ستشتري رمزًا واحدًا لشخص آخر، فيرجى المواصلة ﻹتمام الطلبية. أما إن كنت أنت المتعلم، ف{link}.",
  "payment.messages.enrollment-code-product-info.link": "انفر هنا للتسجيل مباشرة",
  "payment.messages.enrollment.code.product.info.quantity.updated.header": "لقد حدّثنا كميتك.",
  "payment.messages.enrollment.code.product.info.quantity.updated.body": "في سلتك {quantity, plural,\n   one {رمز تسجيل واحد}\n   two {رمزان للتسجيل}\n   few {# رموز تسجيل}\n   many {# رمز تسجيل}\n   other {# رمز تسجيل}\n} تقدّر تكلفتها بـ {price}, ستصلك عبر البريد الإلكتروني.",
  "payment.messages.transaction.declined.body": "لم نستطع معالجة دفعتك، رجاءً تحقق من معلوماتك الخاصة بالدفع أو تقرب من مصرفك أو مؤسستك المالية من أجل المساعدة.",
  "payment.messages.transaction.error.basket_changed": "Your cart has changed since navigating to this page. Please reload the page and verify the product you are purchasing.",
  "payment.messages.key.timeout.body": "من أجل الأمان، ستحتاج مجددًا لإدخال معلومات بطاقة ائتمانك لإتمام الشراء.",
  "payment.messages.key.timeout.2mins.header": "يرجى إتمام الشراء في دقيقتين.",
  "payment.messages.key.timeout.1min.header": "يرجى إتمام الشراء في دقيقة واحدة.",
  "payment.bulk.summary.table.label.price": "السعر",
  "payment.bulk.summary.table.label.quantity": "الكمّية",
  "payment.bulk.summary.table.label.subtotal": "المجموع الفرعي",
  "payment.screen.reader.cart.details.loaded": "يتم الآن تحميل تفاصيل سلّة التسوّق.",
  "payment.section.cart.label": "تفاصيل سلّة التسوق",
  "payment.productlineitem.purchase.cart.heading": "في سلتك",
  "payment.productlineitem.purchase.cart.subheading": "تتضمن مشترياتك ما يلي:",
  "payment.screen.reader.cart.details.loading": "التحميل جارٍ، يرجى الانتظار...",
  "payment.coupon.label": "إضافة رمز قسيمة (اختياري)",
  "payment.coupon.submit": "تطبيق",
  "payment.coupon.benefit.absolute": "تم تطبيق القسيمة {code} لخصم {amount}",
  "payment.coupon.benefit.percentage": "تم تطبيق القسيمة {code} لخصم {amount}‏%",
  "payment.coupon.benefit.default": "تم تطبيق القسيمة {code}",
  "payment.coupon.remove": "إزالة",
  "payment.currency.disclaimer": "* يجتوي هذا المجموع تحويلًا تقريبا، ستدفع {actualAmount} {actualCurrencyCode}.",
  "payment.summary.discount.offer": "خصم {benefit} مقدم من {provider}.",
  "payment.summary.discount.dynamic_offer": "تم تطبيق خصم {benefit} لقاء ترقيتك الأولى.",
  "payment.summary.table.label.discount.total": "الخصوم المطبقة",
  "payment.order.details.course.entitlement": "عند إتمام طلبيتك ستكون قادرًا على تحديد تواريخ المساق من لوحة معلوماتك.",
  "payment.order.details.course.seat.verified": "The above total includes any applicable taxes. After you complete your order you will be automatically enrolled in the verified track of the course.",
  "payment.order.details.course.seat.credit": "عند إتمام طلبيتك ستحصل على اعتماد لمساقك.",
  "payment.order.details.course.seat": "عند استكمال طلبيتك سيتم إلحاقك تلقائيًا في للمساق.",
  "payment.order.details.enrollment.code.terms": "عند إتمام الطلب، فإنك توافق أنت ومؤسستك على الشروط التالية:",
  "payment.order.details.enrollment.code.first.term": "كل رمز يشمل مساقًا واحدًا، ويمكن استخدامه مرة واحدة فقط.",
  "payment.order.details.enrollment.code.second.term": "أنت مسؤل عن توزيع رموز القسائم لمتعلّميك في مؤسستك",
  "payment.order.details.enrollment.code.third.term": "ستنتهي صلاحية كل رمز خلال سنة من تاريخ الشراء أو عند إغلاق المساق -على الأقرب.",
  "payment.order.details.enrollment.code.fourth.term": "إذا لم يتم ضبط إعداد المساق بنظام التعلم الذاتي، فيجب أن تتأكد من توفر مشغل المساق قبل انتهاء صلاحيته.",
  "payment.order.details.enrollment.code.fifth.term": "لا يحق لك إعادة بيع رموز القسائم لأطراف خارجية.",
  "payment.order.details.enrollment.code.sixth.term": "جميع عمليات edX هي عمليات تجارية نهائية وغير مخولة لاسترداد الرسوم.",
  "payment.order.details.enrollment.code.receive.email": "ستتلقى رسالة بريد إلكتروني على {userEmail} برموز التحاقك.",
  "payment.order.details.heading": "ملخّص",
  "payment.productlineitem.professional.certificate": "شهادة مهنية",
  "payment.productlineitem.verified.certificate": "شهادة معتمدة",
  "payment.summary.table.label.price": "السعر",
  "payment.summary.table.label.total.to.pay": "إجمالي",
  "payment.update.quantity.label": "الكمّية",
  "payment.update.quantity.field.help.text": "الحد الأقصى: {number}",
  "payment.update.quantity.submit.button": "تحديث",
  "payment.select.payment.method.heading": "اختر طريقة الدفع",
  "payment.section.payment.details.label": "تفاصيل الدفع",
  "payment.page.method.type.credit": "بطاقة الائتمان",
  "payment.page.method.type.paypal": "باي بال",
  "payment.form.submit.button.text": "إيداع الطلبية",
  "payment.card.details.billing.information.heading": "معلومات الفوترة (مطلوبة)",
  "payment.card.details.expiration.month.label": "شهر انتهاء الصلاحية (مطلوب)",
  "payment.card.details.expiration.year.label": "سنة انتهاء الصلاحية (مطلوب)",
  "payment.page.card.details.month": "الشهر",
  "payment.page.card.details.year": "السنة",
  "payment.card.holder.information.postal.code.label.required": "الرمز البريدي (مطلوب)",
  "payment.card.holder.information.postal.code.label": "الرمز البريدي",
  "payment.card.holder.information.heading": "معلومات حامل البطاقة",
  "payment.card.holder.information.first.name.label": "الاسم الأول (مطلوب)",
  "payment.card.holder.information.last.name.label": "الاسم الأخير (مطلوب)",
  "payment.card.holder.information.organization.label": "المؤسسة (مطلوب)",
  "payment.card.holder.information.address.label": "العنوان (مطلوب)",
  "payment.card.holder.information.unit.label": "رقم الجناح/الشقة",
  "payment.card.holder.information.city.label": "المدينة (مطلوب)",
  "payment.card.holder.information.country.label": "الدولة (مطلوب)",
  "payment.card.holder.information.purchased.for.organization": "أشتري نيابة عن مديري أو أي شخص في مؤسستي المهنية.",
  "payment.card.holder.information.country.options.empty": "اختر البلد",
  "payment.card.details.number.label": "رقم البطاقة (مطلوب)",
  "payment.card.details.security.code.label": "رمز الأمان (مطلوب)",
  "payment.card.details.security.code.help.text": "آخر ثلاثة أرقام في منطقة التوقيع الموجودة على ظهر البطاقة الخاصة بك. بالنسبة لبطاقة أمريكان إكسبرس، يوجد أربعة أرقام على وجه البطاقة.",
  "payment.form.errors.invalid.card.number": "رقم بطاقة غير صحيح",
  "payment.form.errors.unsupported.card": "نوع بطاقة غير مدعوم",
  "payment.form.errors.invalid.security.code": "رمز أمان غير صحيح",
  "payment.form.errors.card.expired": "البطاقة منتهية الصلاحية",
  "payment.form.errors.required.field": "هذا الحقل مطلوب",
  "payment.form.errors.ascii.name": "نأسف للإزعاج، لكن حقل الاسم يقبل حروف ASCII فقط. نحن نعمل على معالجة هذا الأمر ونقدر سعة صدرك.",
  "payment.card.holder.information.state.label": "الولاية/المقاطعة (مطلوبة)",
  "payment.card.holder.information.state.required.label": "الولاية/المقاطعة",
  "payment.card.holder.information.state.options.empty": "اختر الولاية/المنطقة",
  "payment.empty.basket.heading": "سلتك فارغة.",
  "payment.empty.basket.message": "إذا حاولت إجراء عملية شراء، فلن يتم فرض رسوم عليك. عُد إلى {actionLinkOne} للمحاولة مرة أخرى، أو {actionLinkTWO}.",
  "payment.empty.basket.dashboardURL": "لوحة معلومات",
  "payment.empty.basket.supportURL": "اتصل بفريق edX لدعم التجارة الإلكترونية",
  "payment.apple.pay.pay.with.apple.pay": "الدفع بواسطة  Apple Pay",
  "payment.type.paypal": "باي بال",
  "payment.heading.page": "الدفع",
  "payment.loading.payment": "جاري تحميل السلّة...",
  "payment.loading.error": "خطأ: {error}",
  "payment.apple.pay.merchant.validation.failure": "Apple Pay غير متوفر في هذا الوقت. الرجاء تجربة طريقة دفع أخرى.",
  "payment.apple.pay.authorization.failure": "حدث خطأ أثناء معالجة عملية الدفع. لم يتم تحصيل رسوم منك. الرجاء إعادة المحاولة، أو اختيار طريقة أخرى للدفع.",
  "subscription.alerts.error.embargo": "We're sorry, this program is not available in your region.",
  "subscription.alerts.error.program_unavailable": "Something went wrong, please reload the page. If the issue persists please {supportLink}.",
  "subscription.alerts.error.ineligible_program": "We're sorry, this program is no longer offering a subscription option. Please search our catalog for current availability",
  "subscription.checkout.payment.label": "Subscription payment details.",
  "subscription.checkout.billing.notification": "You will be charged {price} {currency} {trialEnd} then monthly until you cancel your subscription.",
  "subscription.checkout.billing.trial.date": "on {date},",
  "subscription.checkout.billing.resubscribe.date": "today,",
  "subscription.checkout.form.submit.button.text.default": "Subscribe",
  "subscription.checkout.form.submit.button.text.resubscribe": "Subscription started",
  "subscription.confirmation.modal.navigation.title": "Go to dashboard",
  "subscription.confirmation.modal.trialing.heading": "Congratulations! Your 7-day free trial of {programTitle} has started.",
  "subscription.confirmation.modal.trialing.body": "When your free trial ends, your subscription will begin, and we'll charge your payment method on file {price} {currency} per month. To avoid being charged, you must cancel before your trial expires. This subscription will automatically renew every month unless you cancel from the {ordersAndSubscriptionLink} page.",
  "subscription.confirmation.modal.resubscribe.heading": "Congratulations! Your subscription to {programTitle} has started.",
  "subscription.confirmation.modal.resubscribe.body": "We charged your payment method {price} {currency}. This subscription will be automatically renewed and charged monthly unless you cancel from the {ordersAndSubscriptionLink} page.",
  "subscription.confirmation.modal.body.orders.link": "Orders and Subscriptions",
  "subscription.purchase.details.heading": "In your cart",
  "subscription.purchase.details.label": "Subscription",
  "subscription.purchase.details.product.list.heading": "Included with your subscription:",
  "subscription.details.order.legal": "By signing up for a subscription or starting a free trial, you authorize us to charge your card on file {price}/month {currency} for {programTitle} and any applicable taxes. Your {programTitle} subscriptions will automatically renew until you cancel. Monthly subscriptions can be canceled at any time, review the cancellation process on the {supportLink}. Canceling a subscription stops the monthly recurring subscription charge, but doesn't refund the transaction for the current billing period. You can continue to access the subscription until the end of the then current billing period.",
  "subscription.details.order.legal.link": "Learner Help Center",
  "subscription.order.details": "After you complete your order you will have verified access to each course in {programTitle}.",
  "subscription.order.details.heading": "Order details",
  "subscription.order.details.taxes": "The above total includes any applicable taxes.",
  "subscription.order.details.cancel": "Cancel anytime.",
  "subscription.details.program.type.microMasters": "MicroMasters",
  "subscription.details.program.type.microBachelors": "MicroBachelors",
  "subscription.details.program.type.xSeries": "XSeries",
  "subscription.details.program.type.professionalProgram": "Professional Program",
  "subscription.details.program.type.professionalCertificate": "Professional Certificate",
  "subscription.details.program.type.masters": "Masters",
  "subscription.screen.reader.details.loading": "Loading, please wait...",
  "subscription.screen.reader.details.loaded": "Shopping details are loaded.",
  "subscription.summary.table.label.price": "Price",
  "subscription.summary.table.label.trialing.price": "/month USD after 7-day free trial",
  "subscription.summary.table.label.resubscribe.price": "/month USD",
  "subscription.summary.table.label.total.to.pay": "Today's total",
  "subscription.heading.page": "Subscription Payment",
  "subscription.loading.details": "Loading subscription..."
}