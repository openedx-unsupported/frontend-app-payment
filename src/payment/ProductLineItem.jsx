import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, injectIntl } from '@edx/frontend-i18n';


class ProductLineItem extends React.PureComponent {
  renderSeatType(seatType) {
    switch (seatType) {
      case 'professional':
      case 'no-id-professional':
        return (<FormattedMessage
          id="payment.productlineitem.professional.certificate"
          defaultMessage="Professional Certificate"
          description="Course certificate type on product details section"
        />);
      case 'verified-certificate':
        return (<FormattedMessage
          id="payment.productlineitem.verified.certificate"
          defaultMessage="Verified Certificate"
          description="Course certificate type on product details section"
        />);
      case 'honor':
      case 'audit':
      default:
        return null;
    }
  }

  render() {
    const {
      imgUrl,
      name,
      seatType,
    } = this.props;
    return (
      <div className="pb-3">
        <h2 className="h6">
          <FormattedMessage
            id="payment.productlineitem.purchase.cart.heading"
            defaultMessage="In your cart"
            description="Heading of the cart in product details section"
          />
        </h2>
        <p>
          <FormattedMessage
            id="payment.productlineitem.purchase.cart.subheading"
            defaultMessage="Your purchase contains the following:"
            description="Subheading of the cart in product details section"
          />
        </p>
        <div className="row align-items-center">
          <div className="col-5">
            <img className="img-thumbnail mb-3" src={imgUrl} alt={name} />
          </div>
          <div className="col-7">
            <p className="h6">
              {name}
            </p>
            <p>{this.renderSeatType(seatType)}</p>
          </div>
        </div>
      </div>
    );
  }
}

ProductLineItem.propTypes = {
  imgUrl: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  seatType: PropTypes.oneOf(['professional', 'no-id-professional', 'verified-certificate', 'honor', 'audit']),
};

ProductLineItem.defaultProps = {
  seatType: undefined,
};

export default injectIntl(ProductLineItem);
