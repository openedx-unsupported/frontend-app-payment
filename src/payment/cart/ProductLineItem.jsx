import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage, FormattedHTMLMessage, injectIntl } from '@edx/frontend-i18n';


class ProductLineItem extends React.PureComponent {
  renderEnrollmentCount(courseKey, enrollmentCountData) {
    const courseObj = enrollmentCountData.find(course => course.key === courseKey);
    if (courseObj) {
      const enrollmentCount = courseObj.enrollment_count;
      return (
        <FormattedHTMLMessage
          id="payment.productlineitem.label.numberEnrolled"
          description="Number of learners enrolled in all of the course runs of this course."
          defaultMessage={'<p class="num-enrolled font-weight-bold">{enrollmentCount} already enrolled!</p>'}
          values={enrollmentCount}
        />
      );
    }
    return null;
  }

  renderCertificateType(certificateType) {
    switch (certificateType) {
      case 'professional':
      case 'no-id-professional':
        return (<FormattedMessage
          id="payment.productlineitem.professional.certificate"
          defaultMessage="Professional Certificate"
          description="Course certificate type on product details section"
        />);
      case 'verified':
        return (<FormattedMessage
          id="payment.productlineitem.verified.certificate"
          defaultMessage="Verified Certificate"
          description="Course certificate type on product details section"
        />);
      case 'honor':
      case 'audit':
      default:
        return null;
    }
  }

  render() {
    const {
      isNumEnrolledExperiment,
      enrollmentCountData,
      imageUrl,
      title,
      certificateType,
      courseKey,
    } = this.props;

    return (
      <div className="row align-items-center mb-3">
        <div className="col-5">
          <div className="embed-responsive embed-responsive-16by9">
            <img className="img-thumbnail product-thumbnail" src={imageUrl} alt="" />
          </div>
        </div>
        <div className="col-7">
          <h6 className="m-0" aria-level="3">{title}</h6>
          <p className="m-0">{this.renderCertificateType(certificateType)}</p>
          {isNumEnrolledExperiment ?
            this.renderEnrollmentCount(courseKey, enrollmentCountData) : null
          }
        </div>
      </div>
    );
  }
}

ProductLineItem.propTypes = {
  isNumEnrolledExperiment: PropTypes.bool,
  enrollmentCountData: PropTypes.arrayOf(PropTypes.shape({
    key: PropTypes.string,
    enrollment_count: PropTypes.number,
  })),
  imageUrl: PropTypes.string,
  title: PropTypes.string,
  certificateType: PropTypes.oneOf(['audit', 'honor', 'verified', 'no-id-professional', 'professional', 'credit']),
  courseKey: PropTypes.string,
};

ProductLineItem.defaultProps = {
  isNumEnrolledExperiment: false,
  enrollmentCountData: null,
  certificateType: undefined,
  title: null,
  imageUrl: null,
  courseKey: null,
};

export default injectIntl(ProductLineItem);
