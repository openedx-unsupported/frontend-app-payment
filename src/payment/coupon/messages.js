import { defineMessages } from '@edx/frontend-i18n';

const messages = defineMessages({
  'payment.coupon.label': {
    id: 'payment.coupon.label',
    defaultMessage: 'Add coupon code (optional)',
    description: 'Label for the add coupon form',
  },
  'payment.coupon.submit': {
    id: 'payment.coupon.submit',
    defaultMessage: 'Apply',
    description: 'Submit button for the add coupon form',
  },
  'payment.coupon.remove': {
    id: 'payment.coupon.remove',
    defaultMessage: 'Remove',
    description: 'Submit button to remove a coupon',
  },
  'payment.coupon.benefit_value': {
    id: 'payment.coupon.benefit_value',
    defaultMessage: 'Coupon {code} applied for {value} off',
    description: 'The description of a coupon for a percentage off the total price. Percentage symbol is included in {value}',
  },
  'payment.coupon.benefit.default': {
    id: 'payment.coupon.benefit.default',
    defaultMessage: 'Coupon {code} applied',
    description: 'The description of a coupon without any specific monetary value off.',
  },
});

export default messages;
