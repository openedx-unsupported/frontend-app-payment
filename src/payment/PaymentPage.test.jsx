/* eslint-disable global-require */
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { IntlProvider, configure as configureI18n } from '@edx/frontend-i18n';
import configureMockStore from 'redux-mock-store';

import * as analytics from '@edx/frontend-analytics';
import { ConnectedPaymentPage } from './';
import { configuration } from '../environment';
import messages from '../i18n';

import ProductLineItem from './ProductLineItem';

const mockStore = configureMockStore();
const storeMocks = {
  defaultState: require('./__mocks__/defaultState.mockStore.js'),
  loading: require('./__mocks__/loading.mockStore.js'),
  loadedBasket: require('./__mocks__/loadedBasket.mockStore.js'),
  loadedFreeBasket: require('./__mocks__/loadedFreeBasket.mockStore.js'),
  loadedEmptyBasket: require('./__mocks__/loadedEmptyBasket.mockStore.js'),
  loadedBasketWithNoTotals: require('./__mocks__/loadedBasketWithNoTotals.mockStore.js'),
};
const requirePaymentPageProps = {
  fetchBasket: () => {},
};

// Mock language cookie
Object.defineProperty(global.document, 'cookie', {
  writable: true,
  value: `${configuration.LANGUAGE_PREFERENCE_COOKIE_NAME}=en`,
});

configureI18n(configuration, messages);

describe('<PaymentPage />', () => {
  describe('Renders correctly in various states', () => {
    it('should render its default state', () => {
      analytics.sendTrackingLogEvent = jest.fn();
      const tree = renderer
        .create((
          <IntlProvider locale="en">
            <Provider store={mockStore(storeMocks.defaultState)}>
              <ConnectedPaymentPage {...requirePaymentPageProps} />
            </Provider>
          </IntlProvider>
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render its loading state', () => {
      analytics.sendTrackingLogEvent = jest.fn();
      const tree = renderer
        .create((
          <IntlProvider locale="en">
            <Provider store={mockStore(storeMocks.loading)}>
              <ConnectedPaymentPage {...requirePaymentPageProps} />
            </Provider>
          </IntlProvider>
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render the basket', () => {
      analytics.sendTrackingLogEvent = jest.fn();
      const tree = renderer
        .create((
          <IntlProvider locale="en">
            <Provider store={mockStore(storeMocks.loadedBasket)}>
              <ConnectedPaymentPage {...requirePaymentPageProps} />
            </Provider>
          </IntlProvider>
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should successfully render a basket with no order totals', () => {
      analytics.sendTrackingLogEvent = jest.fn();
      const tree = renderer
        .create((
          <IntlProvider locale="en">
            <Provider store={mockStore(storeMocks.loadedBasketWithNoTotals)}>
              <ConnectedPaymentPage {...requirePaymentPageProps} />
            </Provider>
          </IntlProvider>
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render an empty basket', () => {
      analytics.sendTrackingLogEvent = jest.fn();
      const tree = renderer
        .create((
          <IntlProvider locale="en">
            <Provider store={mockStore(storeMocks.loadedEmptyBasket)}>
              <ConnectedPaymentPage {...requirePaymentPageProps} />
            </Provider>
          </IntlProvider>
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('should render a free basket', () => {
      analytics.sendTrackingLogEvent = jest.fn();
      const tree = renderer
        .create((
          <IntlProvider locale="en">
            <Provider store={mockStore(storeMocks.loadedFreeBasket)}>
              <ConnectedPaymentPage {...requirePaymentPageProps} />
            </Provider>
          </IntlProvider>
        ))
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});

const product = {
  imageUrl: 'https://prod-discovery.edx-cdn.org/media/course/image/21be6203-b140-422c-9233-a1dc278d7266-941abf27df4d.small.jpg',
  title: 'Introduction to Happiness',
  certificateType: 'verified',
  productType: 'Seat',
  sku: '8CF08E5',
};

describe('<ProductLineItem />', () => {
  describe('Rendering', () => {
    it('should render the product details', () => {
      const tree = renderer.create((
        <IntlProvider locale="en">
          <ProductLineItem {...product} />
        </IntlProvider>
      )).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render the product details for professional certificate', () => {
      product.certificateType = 'professional';
      const tree = renderer.create((
        <IntlProvider locale="en">
          <ProductLineItem {...product} />
        </IntlProvider>
      )).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render the product details for no-id-professional certificate', () => {
      product.certificateType = 'no-id-professional';
      const tree = renderer.create((
        <IntlProvider locale="en">
          <ProductLineItem {...product} />
        </IntlProvider>
      )).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render the product details for verified certificate', () => {
      product.certificateType = 'verified';
      const tree = renderer.create((
        <IntlProvider locale="en">
          <ProductLineItem {...product} />
        </IntlProvider>
      )).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render the product details for unknown seat type', () => {
      product.certificateType = null;
      const tree = renderer.create((
        <IntlProvider locale="en">
          <ProductLineItem {...product} />
        </IntlProvider>
      )).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render the product details for honor certificate', () => {
      product.certificateType = 'honor';
      const tree = renderer.create((
        <IntlProvider locale="en">
          <ProductLineItem {...product} />
        </IntlProvider>
      )).toJSON();
      expect(tree).toMatchSnapshot();
    });
    it('should render the product details for audit certificate', () => {
      product.certificateType = 'audit';
      const tree = renderer.create((
        <IntlProvider locale="en">
          <ProductLineItem {...product} />
        </IntlProvider>
      )).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
