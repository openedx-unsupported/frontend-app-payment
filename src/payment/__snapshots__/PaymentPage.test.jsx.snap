// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentPage /> Renders correctly in various states should render errors 1`] = `
<div
  className="page__payment container-fluid py-5"
>
  <h1>
    Payment
  </h1>
  <div>
    Error: Something went really wrong.
  </div>
</div>
`;

exports[`<PaymentPage /> Renders correctly in various states should render its default state 1`] = `
<div
  className="page__payment container-fluid py-5"
>
  <h1>
    Payment
  </h1>
  <div
    className="row"
  >
    <div
      className="col-6"
    >
      <p>
        Your basket is empty.
      </p>
    </div>
    <div
      className="col-6"
    />
  </div>
</div>
`;

exports[`<PaymentPage /> Renders correctly in various states should render its loading state 1`] = `
<div
  className="page__payment container-fluid py-5"
>
  <h1>
    Payment
  </h1>
  <div>
    <div
      className="d-flex justify-content-center align-items-center flex-column"
      style={
        Object {
          "height": "50vh",
        }
      }
    >
      <div
        className="spinner-border text-primary"
        role="status"
      >
        <span
          className="sr-only"
        >
          Loading basket...
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<PaymentPage /> Renders correctly in various states should render the basket 1`] = `
<div
  className="page__payment container-fluid py-5"
>
  <h1>
    Payment
  </h1>
  <div
    className="row"
  >
    <div
      className="col-6"
    >
      <ul>
        <li>
          paymentProviders: 
          [{"type":"cybersource"},{"type":"paypal"}]
        </li>
        <li>
          sdnCheck: 
          true
        </li>
        <li>
          products: 
          [{"imgUrl":"https://prod-discovery.edx-cdn.org/media/course/image/21be6203-b140-422c-9233-a1dc278d7266-941abf27df4d.small.jpg","name":"Introduction to Happiness","seatType":"verified-certificate"}]
        </li>
      </ul>
      <div
        aria-labelledby="summary-heading"
        aria-live="polite"
        className="basket-section"
        role="region"
      >
        <h2
          className="h6"
          id="summary-heading"
        >
          <span>
            Summary
          </span>
        </h2>
        <table
          className="w-100"
        >
          <tbody>
            <tr>
              <th
                className="font-weight-normal"
                scope="row"
              >
                <span>
                  Price
                </span>
              </th>
              <td
                className="text-right"
              >
                <span>
                  $161.00
                </span>
              </td>
            </tr>
            <tr>
              <th
                className="font-weight-normal"
                scope="row"
              >
                <span>
                  Discounts applied
                </span>
              </th>
              <td
                className="text-right"
              >
                <span>
                  -$12.00
                </span>
              </td>
            </tr>
          </tbody>
        </table>
        <form
          onSubmit={[Function]}
        >
          <div
            className="form-group"
          >
            <label
              className="h6 d-block"
              htmlFor="couponField"
            >
              Add coupon code (optional)
            </label>
            <input
              aria-describedby=""
              className="form-control mb-2"
              id="couponField"
              name="couponField"
              onChange={[Function]}
              type="text"
              value=""
            />
            <button
              className="btn btn-primary"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onKeyDown={[Function]}
              type="submit"
            >
              Apply
            </button>
          </div>
        </form>
        <table
          className="w-100"
        >
          <tbody>
            <tr
              className="font-weight-bold"
            >
              <th
                scope="row"
              >
                <span>
                  TOTAL
                </span>
              </th>
              <td
                className="text-right"
              >
                <span>
                  $149.00
                </span>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        className="basket-section"
      >
        <h2
          className="h6"
        >
          <span>
            Order Details
          </span>
        </h2>
        <p>
          After you complete your order you will be automatically enrolled in the verified track of the course.
        </p>
      </div>
    </div>
    <div
      className="col-6"
    />
  </div>
</div>
`;

exports[`<PaymentPage /> Renders correctly in various states should successfully render a basket with no order totals 1`] = `
<div
  className="page__payment container-fluid py-5"
>
  <h1>
    Payment
  </h1>
  <div
    className="row"
  >
    <div
      className="col-6"
    >
      <ul>
        <li>
          paymentProviders: 
          [{"type":"cybersource"},{"type":"paypal"}]
        </li>
        <li>
          sdnCheck: 
          true
        </li>
        <li>
          products: 
          [{"imgUrl":"https://prod-discovery.edx-cdn.org/media/course/image/21be6203-b140-422c-9233-a1dc278d7266-941abf27df4d.small.jpg","name":"Introduction to Happiness","seatType":"verified-certificate"}]
        </li>
      </ul>
      <div
        aria-labelledby="summary-heading"
        aria-live="polite"
        className="basket-section"
        role="region"
      >
        <h2
          className="h6"
          id="summary-heading"
        >
          <span>
            Summary
          </span>
        </h2>
        <table
          className="w-100"
        >
          <tbody>
            <tr>
              <th
                className="font-weight-normal"
                scope="row"
              >
                <span>
                  Price
                </span>
              </th>
              <td
                className="text-right"
              />
            </tr>
          </tbody>
        </table>
        <table
          className="w-100"
        >
          <tbody>
            <tr
              className="font-weight-bold"
            >
              <th
                scope="row"
              >
                <span>
                  TOTAL
                </span>
              </th>
              <td
                className="text-right"
              />
            </tr>
          </tbody>
        </table>
      </div>
      <div
        className="basket-section"
      >
        <h2
          className="h6"
        >
          <span>
            Order Details
          </span>
        </h2>
        <p>
          After you complete your order you will be automatically enrolled in the verified track of the course.
        </p>
      </div>
    </div>
    <div
      className="col-6"
    />
  </div>
</div>
`;
